src/app/portlets/dot-edit-page/content/components/dot-edit-page-toolbar/dot-edit-page-toolbar.component.spec.ts

```diff
@@ -627,4 +620,52 @@ describe('DotEditPageToolbarComponent', () => {
             expect(dotDialogService.confirm).not.toHaveBeenCalled();
         });
     });
+
+    describe('lock page state', () => {
+
+        beforeEach(() => {
+            fixture.componentInstance.pageState.page.canLock = true;
+            fixture.componentInstance.pageState.state.locked = true;
+            fixture.componentInstance.pageState.page.lockedBy = 'someone';
+        });
+
+        it('should have page is locked by another user message and not disabled edit button', () => {
+            fixture.componentInstance.pageState.state.lockedByAnotherUser = true;
+
+            fixture.detectChanges();
+
+            const lockedMessage: DebugElement = de.query(By.css('.edit-page-toolbar__locked-by-message'));
+            expect(lockedMessage.nativeElement.textContent).toContain('Page is locked');
+
+            const editStateModel = component.states.find((state) => state.label === 'Edit');
+            expect(editStateModel.styleClass).toEqual('');
+            expect(component.lockerModel).toBeFalsy();
+        });
+
+        it('should have not show any meesage and not disabled edit button neither when page is locked by current user', () => {
+            fixture.componentInstance.pageState.state.lockedByAnotherUser = false;
+            fixture.componentInstance.pageState.page.canLock = true;
```

Comment:
is canLock, locked and lockedBy really needed since are present in the before each ?